source/simulator/solver.cc

```diff
@@ -168,7 +168,7 @@ namespace aspect
     /**
      * Implement the block Schur preconditioner for the Stokes system.
      */
-    template <class PreconditionerA, class PreconditionerMp>
+    template <class PreconditionerA, class SInvOperator>
```

Comment:
Is the `Preconditioner` part of the name not appropriate? Would it then not be consistent to change both names as in:
```suggestion
    template <class AOperator, class SInvOperator>
```

Or could it be like the following:
```suggestion
    template <class PreconditionerA, class PreconditionerSInv>
```

I am just asking if it is worth to preserve the symmetry of the names.

Reply:
I think what @gassmoeller was asking is why there is now in imbalance in the naming of the two arguments. The first is called a preconditioner, the second an operator. Is that because the second is an exact representation of the inverse of S? I don't think that's what you are providing. Perhaps a better choice would be to call these two arguments `AinvApproximation` and `SinvApproximation` or some such, to indicate that what that type represents is some kind of approximation to the inverse of the matrix?